src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp

```diff
@@ -86,7 +93,10 @@ ov::pass::ReshapeAMatMul::ReshapeAMatMul() {
 
 ov::pass::ReshapeBMatMul::ReshapeBMatMul() {
     MATCHER_SCOPE(ReshapeBMatMul);
-    auto other_input_label = pattern::any_input();
+    auto other_input_predicate = [](ov::Output<ov::Node> output) -> bool {
+        return output.get_partial_shape().rank().is_static();
+    };
+    auto other_input_label = pattern::any_input(other_input_predicate);
```

Comment:
I think we should make use of what @jane-intel has suggested before.

Reply:
done